src/s3/commands/presignedURL.ts

```diff
@@ -38,36 +45,124 @@ export async function presignedURLCommand(
+        this.form.signedUrlParams.operation.bindPrompter(() => createOperationPrompter())
+        this.form.signedUrlParams.bucketName.bindPrompter(({ region }) =>
+            createBucketPrompter(assertDefined(region, 'region'))
+        )
+
+        if (node) {
+            this.form.signedUrlParams.key.setDefault(node.file.key)
+            this.form.signedUrlParams.key.bindPrompter(
+                ({ region, signedUrlParams }) =>
+                    createS3FilePrompter(
+                        assertDefined(region, 'region'),
+                        assertDefined(signedUrlParams?.bucketName, 'bucketName'),
+                        assertDefined(signedUrlParams?.operation, 'operation')
+                    ),
+                { showWhen: state => state.signedUrlParams?.operation === 'putObject' }
+            )
+        } else {
+            this.form.signedUrlParams.key.bindPrompter(({ region, signedUrlParams }) =>
+                createS3FilePrompter(
+                    assertDefined(region, 'region'),
+                    assertDefined(signedUrlParams?.bucketName, 'bucketName'),
+                    assertDefined(signedUrlParams?.operation, 'operation')
+                )
+            )
+        }
 
-    const time = Number(timeStr)
+        this.form.signedUrlParams.time.bindPrompter(() => createExpiryPrompter().transform(s => Number(s) * 60))
 
-    if (isNaN(time) || time < 0) {
-        throw new ToolkitError('The provided expiration time is not a positive number', { code: 'InvalidInput' })
+        function assertDefined<T>(val: T | undefined, key: string): T {
+            if (val === undefined) {
+                throw Error(`PresignedUrlWizard: "${key}" is undefined`)
+            }
+            return val
+        }
     }
+}
 
-    return time
+function createOperationPrompter() {
+    const items: DataQuickPickItem<string>[] = [
+        { label: 'Download (GET)', data: 'getObject' },
+        { label: 'Upload (PUT)', data: 'putObject' },
+    ]
+
+    return createQuickPick(items, { title: 'Presigned URL: Choose an operation', buttons: createCommonButtons() })
 }
 
-function validateTime(time: string): string | undefined {
-    const number = Number(time)
+function createExpiryPrompter() {
+    return createInputBox({
+        value: '15',
+        prompt: 'Specify the expiry time (minutes) for the URL',
+        placeholder: 'Defaults to 15 minutes',
+        validateInput: validateTime,
+        buttons: createCommonButtons(),
+    })
+}
 
-    if (isNaN(Number(time)) || !Number.isSafeInteger(number) || Number(time) <= 0) {
-        return invalidNumberWarning
+function createBucketPrompter(region: string) {
+    const client = new DefaultS3Client(region)
+    const items = client.listBucketsIterable().map(b => [
+        {
+            label: b.Name,
+            data: b.Name,
+        },
+    ])
+
+    return createQuickPick(items, { title: 'Select an S3 Bucket', buttons: createCommonButtons() })
```

Comment:
should default to recently-used bucket. some sample code is here: https://github.com/aws/aws-toolkit-vscode/blob/e5f813fb84c7f0de5706d72acce8f1e982767588/src/shared/sam/sync.ts#L57

Reply:
This will be implemented in an upcoming change.